SpatialGDKEditor/Source/Private/SpatialGDKEditor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditor.h"
+
+#include "Async.h"
+#include "SpatialGDKEditorSchemaGenerator.h"
+
+#include "Editor.h"
+
+#include "AssetRegistryModule.h"
+#include "GeneralProjectSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditor);
+
+USpatialGDKEditor::USpatialGDKEditor()
+	: bSchemaGeneratorRunning(false)
+{
+
+}
+
+void USpatialGDKEditor::GenerateSchema(FSimpleDelegate SuccessCallback, FSimpleDelegate FailureCallback, FSpatialGDKEditorErrorHandler ErrorCallback)
+{
+	if (bSchemaGeneratorRunning)
+	{
+		UE_LOG(LogSpatialGDKEditor, Warning, TEXT("Schema generation is already running"));
+		return;
+	}
+
+	bSchemaGeneratorRunning = true;
+
+	// Force spatial networking so schema layouts are correct
+	UGeneralProjectSettings* GeneralProjectSettings = GetMutableDefault<UGeneralProjectSettings>();
+	bool bCachedSpatialNetworking = GeneralProjectSettings->bSpatialNetworking;
+	GeneralProjectSettings->bSpatialNetworking = true;
+
+
+	// Ensure all our spatial classes are loaded into memory before running
+	CacheSpatialObjects(SPATIALCLASS_GenerateTypeBindings, ErrorCallback);
+
+	SchemaGeneratorResult = Async<bool>(EAsyncExecution::Thread, SpatialGDKGenerateSchema,
+		[this, bCachedSpatialNetworking, SuccessCallback, FailureCallback]()
+	{
+		if (!SchemaGeneratorResult.IsReady() || SchemaGeneratorResult.Get() != true)
+		{
+			FailureCallback.Execute();
```

Comment:
Is it ever valid to call this without a failure or success callback? If so, check `FailureCallback.IsBound()` before executing.